### PR TITLE
Add `RollingWAvg` feature to UpdateBy

### PR DESCRIPTION
**Pull request by lbooker42**

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingwavg/BaseBigNumberRollingWAvgOperator.java
Comment (nbauernfeind) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingwavg/BaseBigNumberRollingWAvgOperator.java:82]: I don't think we should use `Basic.isNull` here. This forces the primitive to be boxed. Can we compare against `QueryConstants.NULL_TYPE`  values instead?

File: engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingwavg/BasePrimitiveRollingWAvgOperator.java
Comment (nbauernfeind) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingwavg/BasePrimitiveRollingWAvgOperator.java:71]: Same opinion on the `Basic.isNull` use here. It is significantly cheaper to compare to the `QueryConstants.NULL_TYPE` since we know the type directly. Maybe we should consider adding a `Chunk#isNull(int index)` -- there are probably lots of use cases.

**Comment by lbooker42** [comment]
This needs refactored before another review and merge.  Executive decision to have `double` output for primitive input and weight columns but promote to `BigDecimal` when either input or weight column is `BigInteger`/`BigDecimal`.

File: java-client/session/src/test/java/io/deephaven/client/impl/UpdateBySpecBuilderTest.java
Comment (nbauernfeind) [inline java-client/session/src/test/java/io/deephaven/client/impl/UpdateBySpecBuilderTest.java:203]: missed issue number

**Review by nbauernfeind — APPROVED** [review]

**Comment by deephaven-internal** [comment]
Labels indicate documentation is required. Issues for documentation have been opened:

How-to: https://github.com/deephaven/deephaven.io/issues/2495
Conceptual: https://github.com/deephaven/deephaven.io/issues/2494
Reference: https://github.com/deephaven/deephaven.io/issues/2497
Blog: https://github.com/deephaven/deephaven.io/issues/2496

**Pull request merged**
